// Copyright 2020 The Chromium OS Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto3";
option optimize_for = LITE_RUNTIME;

// This file defines messages used for interacting with the document scanner
// utility, lorgnette.
package lorgnette;

// Describes possible types of sources that can be supported by a scanner.
enum SourceType {
  SOURCE_PLATEN = 0;
  SOURCE_ADF_SIMPLEX = 1;
  SOURCE_ADF_DUPLEX = 2;
}

// A source that can be scanned from for a scanner.
message DocumentSource {
  // The type of this source.
  SourceType type = 1;

  // The name for this source used by the scanner backend.
  string name = 2;
}

// The color modes that may be supported for a particular scanner.
enum ColorMode {
  MODE_LINEART = 0;
  MODE_GRAYSCALE = 1;
  MODE_COLOR = 2;
}

// Information returned from a GetScannerCapabilities dbus request.
message ScannerCapabilities {
  // Discrete scanning resolutions.
  repeated uint32 resolutions = 1;

  // Sources to scan a document from.
  repeated DocumentSource sources = 2;

  // Color modes to use for a document.
  repeated ColorMode color_modes = 3;
}
